src/workers/search/project-search.js

```diff
@@ -4,23 +4,23 @@
 
 // Maybe reuse file search's functions?
 
-export function findSourceMatches(source, queryText) {
-  const text = source.text;
-  if (source.loadedState !== "loaded" || !text || queryText == "") {
+export function findSourceMatches(source = {}, queryText) {
```

Comment:
Not sure if we really want `findSourceMatches` to gracefully handle the lack of a source parameter or just outright throw. I'd say throwing is better for the application in terms of bug hunting and will save development time.

Reply:
Yeah, I think we should throw if there's no source, otherwise we're just sort of gracefully doing work we don't need to be doing.

Reply:
i wouldagree we should not default to an empty object as `findSourceMatches` should always be called with a source. if it is not, it should probably fail . How do we differentiate between an invalid or broken source (e.g source is null ) and a valid source that has no matches ?